controller/Redirector.php

```diff
@@ -15,59 +15,79 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  *
- * Copyright (c) 2018 (original work) Open Assessment Technologies SA ;
+ * Copyright (c) 2018-2022 (original work) Open Assessment Technologies SA.
+ *
+ * @author Gyula Szucs <gyula@taotesting.com>
  */
 
+declare(strict_types=1);
+
 namespace oat\taoBackOffice\controller;
 
+use Throwable;
+use tao_actions_CommonModule;
+use InvalidArgumentException;
+use common_exception_MissingParameter;
 use oat\generis\model\OntologyAwareTrait;
+use oat\tao\model\http\HttpJsonResponseTrait;
 use oat\tao\model\taskQueue\TaskLogInterface;
 use oat\taoBackOffice\model\routing\ResourceUrlBuilder;
 
-/**
- * @author Gyula Szucs <gyula@taotesting.com>
- */
-class Redirector extends \tao_actions_CommonModule
+class Redirector extends tao_actions_CommonModule
 {
-    const PARAMETER_TASK_ID = 'taskId';
-
     use OntologyAwareTrait;
+    use HttpJsonResponseTrait;
+
+    private const PARAMETER_TASK_ID = 'taskId';
 
     /**
      * Redirect to a resource generated in a task.
      */
-    public function redirectTaskToInstance()
+    public function redirectTaskToInstance(): void
```

Comment:
suggestion: CodeClimate complains about this method being too long, maybe we can move obtaining the entity, the URI and/or the resource to a helper method (sort-of reuqest parser/converter) and keep here just the try/catch

```

public function redirectTaskToInstance(): void
{
    $resource = $this->getInstanceForRedirectRequest($this->getPsrRequest());

    try {
      $this->setSuccessJsonResponse($this->getResourceUrlBuilder()->buildUrl($resource));
    } ....
}
```

```
private function getInstanceForRedirectRequest(Request $request): \oat\tao\model\taskQueue\TaskLog\Entity\EntityInterface
{
        $queryParams = $request->getQueryParams();

    if (!isset($queryParams[self::PARAMETER_TASK_ID])) {
        throw new common_exception_MissingParameter(self::PARAMETER_TASK_ID, $this->getRequestURI());
    }

        $entity = $this->getTaskLog()->getByIdAndUser(
            $queryParams[self::PARAMETER_TASK_ID],
            $this->getSession()->getUserUri(),
            true // in Sync mode, task is archived straightaway
        );

        retrun $this->getResource($entity->getResourceUriFromReport());
}
```

Reply:
It's a bit out of scope for my task 😅 

I will do this if I will have time, thanks 🙂